pkg/apis/kops/validation/validation.go

```diff
@@ -254,6 +254,52 @@ func validateClusterSpec(spec *kops.ClusterSpec, c *kops.Cluster, fieldPath *fie
 		}
 	}
 
+	if spec.IAM != nil {
+		if len(spec.IAM.ServiceAccountExternalPermissions) > 0 {
+			if spec.ServiceAccountIssuerDiscovery == nil || !spec.ServiceAccountIssuerDiscovery.EnableAWSOIDCProvider {
```

Comment:
The intention here was to ensure that AWS OIDC provider has been enabled if user tries to configure external permissions. Technically one should look for the `AWS` substruct of `ServiceAccountExternalPermissions` but since this we only support `AWS` now, we can defer that to later.

Reply:
Sorry, I misread the condition as testing two separate things.